app/models/educator.rb

```diff
@@ -151,6 +151,10 @@ def allowed_sections
     end
   end
 
+  def allowed_sections_index
+    allowed_sections.map { |section| section.id }
```

Comment:
Not sure the value of adding to the `Educator` model here?  You could inline `current_educator_allowed_sections: `educator.allowed_sections.map(&:id)` in the controller or a helper there.